src/patternfly/components/Menu/menu.scss

```diff
@@ -173,23 +169,37 @@
   &.pf-m-drilldown {
     display: flex;
     flex-direction: column;
-    transition: var(--pf-c-menu--m-drilldown--Transition);
```

Comment:
I believe this change will fix the original issue. PopperJS is applying `transform: translate3d(...)` to the outer `.pf-c-menu` to position the menu relative to the toggle, and this transition on `.pf-c-menu.pf-m-drilldown` was causing that position to transition in. You can see it if you add `className="pf-m-drilldown"` to one of the `<Menu>` components in the [composable menu demos](https://www.patternfly.org/v4/demos/composable-menu). 

Unless there's another reason to transition `left` on the inner menus, I think transitioning `translate()` is better because it's more performant.

Reply:
Also an aside, I found this in the popperJS docs - looks like you may be able to change the way popper positions the menu to stick with `top` and `left` positioning if we still have conflicting styles - https://popper.js.org/docs/v2/modifiers/compute-styles/#gpuacceleration.